ocs_ci/ocs/perftests.py

```diff
@@ -70,6 +79,91 @@ def teardown(self):
         if hasattr(self, "operator"):
             self.operator.cleanup()
 
+        now_data = self.get_cephfs_data()
+        # Wait 1 minutes for the backend deletion actually start.
+        log.info("Waiting for Ceph to finish cleaning up")
+        time.sleep(60)
+
+        # Quarry the storage usage every 2 Min. if no difference between two
+        # samples, the backend cleanup is done.
+        still_going_down = True
+        while still_going_down:
+            new_data = self.get_cephfs_data()
+            # no deletion operation is in progress
+            if abs(now_data - new_data) < 1:
+                still_going_down = False
+                # up to 2% inflation of usage is acceptable
+                if new_data > (self.used_capacity * 1.02):
+                    log.warning(
+                        f"usage capacity after the test ({new_data:.2f} GiB) "
+                        f"is more then in the begining of it ({self.used_capacity:.2f} GiB)"
+                    )
+            else:
+                log.info(f"Last usage : {now_data}, Current usage {new_data}")
+                now_data = new_data
+                log.info("Waiting for Ceph to finish cleaning up")
+                time.sleep(120)
+                still_going_down = True
+        log.info("Storage usage was cleandup")
```

Comment:
Apart from +ve scenario there is no other way this loop will break, How about adding a break by checking particular time interval that expected capacity is not reached.

Reply:
@ramkiperiy ,
the braking point of the loop is when there is no deletion operation in the background (line 93), yes it can take a lot of time as you can see in the [validation job log](https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/8873/consoleFull) 
```
2022-01-05 16:52:34  PASSED                                                                   [100%]
2022-01-05 16:52:34  ------------------------------ live log teardown -------------------------------
... lines deleted ...
2022-01-05 16:54:53  14:54:53 - MainThread - ocs_ci.ocs.perftests - INFO - Waiting for Ceph to finish cleaning up
2022-01-05 16:56:02  14:55:53 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage rsh rook-ceph-tools-cdbf6df6b-mqtdm ceph df --format json-pretty
2022-01-05 16:56:02  14:55:54 - MainThread - ocs_ci.ocs.perftests - INFO - Last usage : 1656.673387539573, Current usage 1652.7047748686746
2022-01-05 16:56:02  14:55:54 - MainThread - ocs_ci.ocs.perftests - INFO - Waiting for Ceph to finish cleaning up
... lines deleted ...
2022-01-06 01:02:09  23:02:01 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage rsh rook-ceph-tools-cdbf6df6b-mqtdm ceph df --format json-pretty
2022-01-06 01:02:09  23:02:02 - MainThread - ocs_ci.ocs.perftests - WARNING - usage capacity after the test (99.03 GiB) is more then in the begining of it (0.80 GiB)
2022-01-06 01:02:09  23:02:02 - MainThread - ocs_ci.ocs.perftests - INFO - Storage usage was cleandup
```
as you can see, it took ~8 Hours (this it something i am trying to reduce in the PR), but it exit the loop